### PR TITLE
[RHPAM-4423] Upgrade to Quarkus 2.13.4 and add example for mTLS using vault

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: ruromero <rromerom@redhat.com>

**JIRA**:

[RHPAM-4423](https://issues.redhat.com/browse/RHPAM-4423)

### CODE REVIEW

**Comment by gmunozfe** [comment]
jenkins retest this

File: examples/authentication/mtls/mtls-vault.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      provider:
+        pim:
+          path: config/pimvault.p12
+          secret: ${vault.storepassword} # This will be provided as a property
+  http:
+    ssl:
+      certificate:
+        credentials-provider: quarkus.file.vault.provider.pim
+        key-store-file: config/server-keystore.jks
+        key-store-alias: pim
```
Comment (gmunozfe) [inline examples/authentication/mtls/mtls-vault.yml]:
```suggestion
        key-store-key-alias: pim
```

The other one doesn't exist (or couldn't find it at least at Quarkus documentation)

**Review by gmunozfe — COMMENTED** [review]
Looks good to me, just a comment regarding a variable name in the sample file.
Outstanding work @ruromero !

File: examples/authentication/mtls/mtls-vault.yml (hunk 1 above)
Comment (ruromero) [reply examples/authentication/mtls/mtls-vault.yml]: if it is not the right key, then I should remove it completely from the example as the defaults worked as expected

**Review by gmunozfe — APPROVED** [review]
Great, ready to merge then @mareknovotny

**Comment by mareknovotny** [comment]
wait with merging as next release is still on Quarkus 2.7.x

**Comment by mareknovotny** [comment]
ok, i will merge this, please prepare backports for 7.67.x and 7.67.x-blue branches, but don't merge them yet

**Pull request merged**